paasta_tools/check_autoscaler_max_instances.py

```diff
@@ -100,8 +100,15 @@ async def check_max_instances(
                 >= autoscaling_status["max_instances"]
             ):
                 threshold = job_config.get_autoscaling_max_instances_alert_threshold()
-                setpoint = job_config.get_autoscaling_params()["setpoint"]
-                metric_threshold_target_ratio = threshold / setpoint
+                metrics_provider_configs = job_config.get_autoscaling_params()[
+                    "metrics_providers"
+                ]
+
+                # TODO: this doesn't work for metrics_providers that don't use setpoint (e.g. active-requests)
+                max_setpoint = max(
+                    [provider["setpoint"] for provider in metrics_provider_configs]
+                )
+                metric_threshold_target_ratio = threshold / max_setpoint
```

Comment:
This feels weird to me so I've got a branch (based on this one) that makes max_instances_alert_threshold be per-metrics_provider: https://github.com/Yelp/paasta/pull/3850

Reply:
Cool, this makes more sense than what I was trying to do.  I misunderstood how this threshold was interpreted/applied.